core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/jdbc/JDBCUserStoreManager.java

```diff
@@ -134,6 +138,15 @@ public JDBCUserStoreManager(RealmConfiguration realmConfig, int tenantId) throws
 //		}
 
         // new properties after carbon core 4.0.7 release.
+        Map<String, Object> userStorePropertiesMap = new HashMap<>(realmConfig.getUserStoreProperties());
+        String digestFunction = userStorePropertiesMap.get(JDBCRealmConstants.DIGEST_FUNCTION).toString();
+        HashProviderFactory hashProviderFactory = UserStoreMgtDataHolder.getInstance().getHashProviderFactory(digestFunction);
+        if (hashProviderFactory == null) {
+            hashProvider = null;
+        } else {
+            hashProvider = hashProviderFactory.getHashProvider(userStorePropertiesMap);
+        }
```

Comment:
Can't we directly assign the value to `hashProvider`?

Reply:
No, we need to validate for `hashProviderFactory` being null. Otherwise, it will throw NPE.